syft/pkg/type.go

```diff
@@ -17,6 +17,7 @@ const (
 	JavaPkg          Type = "java-archive"
 	JenkinsPluginPkg Type = "jenkins-plugin"
 	GoModulePkg      Type = "go-module"
+	RustPkg          Type = "rust-crate"
```

Comment:
I struggled with what to name this, so certainly up for any feedback here.  These were the ones I considered:

- rust
- cargo
- rust-cargo
- rust-crate

I ended up with `rust-crate` for the moment because that really describes the equivalent of a package format in rust, whereas `cargo` is the tool for managing crates, and that seemed to fit the best based off of the naming for the others

Reply:
short answer: I think `rust-crate` is a good name 👍 

longer answer:
We struggle with this too with each new package type we add --we've had a lot of internal discussions on what a "package type" is and isn't. We haven't come to any strong conclusions. There are three fields that should be considered together when naming:
1. `Package.Type`, the packaging ecosystem (in this case we're within the `cargo` packaging ecosystem) or the packaging unit (in this case `crate`)
2. `Package.Language`, a definitive language ecosystem the package belongs to (`rust`)
3. `Package.MetadataType`, hints at what the data shape is for the `Package.Metadata` field, which should simply surface raw information from the underlying evidence parsed (in this case, from the `Cargo.lock` file).

Looking at some resources regarding rust packaging (https://rust-cli.github.io/book/tutorial/packaging.html, https://doc.rust-lang.org/cargo/appendix/glossary.html#package, https://github.com/rust-lang/cargo/issues/3380) there seems to be something hinting that the semantics are really about `cargo-packages` instead of `crates`, and to add even more confusion, there is a separate crate file format (https://users.rust-lang.org/t/where-to-learn-about-crate-file-format/11496) that seems to not be talked about much. 

That being said, from a rust-community-perspective, I think that `crate` is the unit that folks are used to thinking in, and cargo is essentially describing. (also, if we're wrong we can change it later to be more accurate)

Reply:
Yes, it is a bit confusing, and I now can't seem to find the reference I had remembered seeing in the rust book about a `crate` being similar to the idea of a package from other ecosystems, but I did find https://doc.rust-lang.org/stable/book/ch01-03-hello-cargo.html, and it mentions:

> The last line, [dependencies], is the start of a section for you to list any of your project’s dependencies. In Rust, packages of code are referred to as crates. We won’t need any other crates for this project, but we will in the first project in Chapter 2, so we’ll use this dependencies section then.

This may just be another example like Steve mentions in https://github.com/rust-lang/cargo/issues/3380#issuecomment-266106939 where it isn't always used in a clear way, or it may be as intended, so when you specify dependencies you are specifying the crate you want, in which case I think we've named it appropriately?  

I also saw https://doc.rust-lang.org/edition-guide/rust-2018/cargo-and-crates-io/cargo-workspaces-for-multi-package-projects.html which seems to summarize the hierarchy fairly well.